### PR TITLE
(maint) Add a function to allow for backwards compatible credentials

### PR DESCRIPTION
**Pull request by willmeek**
This commit will add a function to allow for backwards compatible credentials.
- address  (map to host)
- username (map to user)

### CODE REVIEW

File: lib/puppet_x/puppetlabs/cisco_ios/transport_shim.rb [hunk 1]
```diff
@@ -301,18 +301,11 @@ class Puppet::ResourceApi::Transport::Wrapper
...
   def initialize(name, url_or_config_or_transport)
-    if url_or_config_or_transport.is_a? String
-      url = URI.parse(url_or_config_or_transport)
-      raise "Unexpected url '#{url_or_config_or_transport}' found. Only file:/// URLs for configuration supported at the moment." unless url.scheme == 'file'
-      raise "Trying to load config from '#{url.path}, but file does not exist." if url && !File.exist?(url.path)
-      config = self.class.deep_symbolize(Hocon.load(url.path, syntax: Hocon::ConfigSyntax::HOCON) || {})
-    elsif url_or_config_or_transport.is_a? Hash
-      config = url_or_config_or_transport
-    elsif transport_class?(name, url_or_config_or_transport)
+    if transport_class?(name, url_or_config_or_transport)
```
Comment (Thomas-Franklin) [inline lib/puppet_x/puppetlabs/cisco_ios/transport_shim.rb]: Yea these changes will be breaking for Puppet 6.0.x which this file is meant to make compatible.

Comment (willmeek) [reply lib/puppet_x/puppetlabs/cisco_ios/transport_shim.rb]: I tested against Puppet 6 (6.0.3 and 6.4.2) and 5.5 locally and it functioned.

Tried to keep what I'm doing similar to what was there before.

Comment (Thomas-Franklin) [reply lib/puppet_x/puppetlabs/cisco_ios/transport_shim.rb]: Yes but locally you probably had latest rsapi, on a PE with 6.0.x (2019.0) the install of rsapi is locked to 1.6.x of the rsapi and the required changes are in 1.8.x. so I wouldn't make the changes in this initialize

Comment (Thomas-Franklin) [reply lib/puppet_x/puppetlabs/cisco_ios/transport_shim.rb]: The device.rb changes are acceptable

Comment (willmeek) [reply lib/puppet_x/puppetlabs/cisco_ios/transport_shim.rb]: Added check that the Puppet::Transport module exists before calling this code.

Previously it wasn't hit as the 'am I a hash' / 'am I a string' checks caught the input before it got to this 'if transport_class?' branch

Tested against RSAPI 1.6.4, Puppet 6.0.3
and latest RSAPI 1.8.1, Puppet 6.4.0

File: metadata.json [hunk 2]
```diff
@@ -9,7 +9,7 @@
   "dependencies": [
     {
       "name": "puppetlabs/resource_api",
-      "version_requirement": ">= 1.0.0 < 2.0.0"
+      "version_requirement": ">= 1.6.3 < 2.0.0"
```
Comment (Thomas-Franklin) [inline metadata.json]: 💯 👍

Comment (Thomas-Franklin) [reply metadata.json]: Wait actually, no - this is the resource_api module and not the gem

Comment (Thomas-Franklin) [reply metadata.json]: And only 1.0.0 has been released of it (https://forge.puppet.com/puppetlabs/resource_api) so this needs to be changed back :-(

Comment (Thomas-Franklin) [inline metadata.json]: Message seems to have deleted, but this cannot be updated as this refers to the module to help install the RSAPI on server/proxy and has only 1.0.0 as a released version

Comment (willmeek) [reply metadata.json]: Saw the message in email, reverted this

File: lib/puppet_x/puppetlabs/cisco_ios/transport_shim.rb [hunk 3]
```diff
@@ -301,18 +301,19 @@ class Puppet::ResourceApi::Transport::Wrapper
...
-    if url_or_config_or_transport.is_a? String
-      url = URI.parse(url_or_config_or_transport)
-      raise "Unexpected url '#{url_or_config_or_transport}' found. Only file:/// URLs for configuration supported at the moment." unless url.scheme == 'file'
-      raise "Trying to load config from '#{url.path}, but file does not exist." if url && !File.exist?(url.path)
-      config = self.class.deep_symbolize(Hocon.load(url.path, syntax: Hocon::ConfigSyntax::HOCON) || {})
-    elsif url_or_config_or_transport.is_a? Hash
-      config = url_or_config_or_transport
-    elsif transport_class?(name, url_or_config_or_transport)
-      @transport = url_or_config_or_transport
+    # Check if the Puppet::Transport module exists
```
Comment (Thomas-Franklin) [inline lib/puppet_x/puppetlabs/cisco_ios/transport_shim.rb]: Upon re-looking, this code could just bee removed and the `@transport ||= Puppet::ResourceApi::Transport.connect(name, url_or_config_or_transport)` is enough, as this code will only ever get called in 1.6.x versions of the RSAPI anyway.

**Pull request merged**